xlators/mgmt/glusterd/src/glusterd.c

```diff
@@ -1553,21 +1552,9 @@ init(xlator_t *this)
 #endif
 
 #ifndef GF_DARWIN_HOST_OS
-    {
-        struct rlimit lim;
-        lim.rlim_cur = 65536;
-        lim.rlim_max = 65536;
-
-        if (setrlimit(RLIMIT_NOFILE, &lim) == -1) {
-            gf_smsg(this->name, GF_LOG_ERROR, errno, GD_MSG_SET_XATTR_FAIL,
-                    "Failed to set 'ulimit -n 65536'", NULL);
-        } else {
-            gf_msg(this->name, GF_LOG_INFO, 0, GD_MSG_FILE_DESC_LIMIT_SET,
-                   "Maximum allowed open file descriptors "
-                   "set to 65536");
-        }
-    }
-#endif
+    if (gf_set_nofile(0, 65536) == RLIM_INFINITY)
+        gf_smsg(this->name, GF_LOG_WARNING, errno, LG_MSG_NOFILE_FAILED, NULL);
```

Comment:
I think it would be better to add the log message inside `gf_set_nofile()`. Callers shouldn't care to log the message.

I would also make the function definition conditional, so that on Darwin it's just defined as an empty op (`#define gf_set_nofile(_high, _low) do { } while (0)`), and as a normal function everywhere else. This way we can call it without needing to add the `#ifdef` block each time.

Reply:
OK so if we don't need a special return value to indicate error (and so issue warning), it looks reasonable convert `gf_set_nofile()` to plain `void`.